cfe_internal/enterprise/federation/federation.cf

```diff
@@ -757,7 +757,7 @@ bundle agent superhub_schema
       "$(cfengine_enterprise_federation:config.bin_dir)/psql_wrapper.sh"
         arglist => {
                      "cfdb",
-                     `"select superhub_schema('$(sys.key_digest)');"`,
+                     "select superhub_schema('$(sys.key_digest)');",
```

Comment:
Assuming this change is done for 3.24.0 ... When I upgrade the first thing I will do is upgrade my policy set. So I will be running e.g. 3.18.6 with this policy for some time before I upgrade to 3.24.0.

Ideally we could handle this automatically. Seems like it could be difficult to detect reliably, but this is not an elegant transition.

People will have to duplicate all the cases where they have double quoting in argslist where the behavior changes and make sure that policy uses the right quoting based on the cfengine version.

Reply:
Yes, it would be a breaking change. Do you think we should not make it?

Reply:
Ping @nickanderson

Reply:
Do I understand correctly? ...

- Users must locate all use of `argslist` within their policy
- Users must add the macro treatment (or similar) as done in this file **if** the promise using `argslist` does **not** execute within a shell.
- Users must NOT add the macro treatement (or similar) if the promise using `argslist` uses a shell

Am I overlooking anything?

Reply:
Sounds about right :)

Reply:
Ok, we just need a good change in behavior post to outline the details I guess.